src/core/Akka.Remote/Transport/ThrottleTransportAdapter.cs

```diff
@@ -547,33 +544,22 @@ private Task<SetThrottleAck> AskModeWithDeathCompletion(IActorRef target, Thrott
             if (target.IsNobody()) return Task.FromResult(SetThrottleAck.Instance);
             else
             {
-                return target.Ask<SetThrottleAck>(mode, timeout);
+                //return target.Ask<SetThrottleAck>(mode, timeout);
 
                 //TODO: use PromiseActorRef here when implemented
-                //var internalTarget = target.AsInstanceOf<InternalActorRef>();
-                //var promiseRef = PromiseActorRef.Apply(internalTarget.Provider, timeout, target, mode.GetType().Name);
-                //internalTarget.Tell(new Watch(internalTarget, promiseRef));
-                //target.Tell(mode, promiseRef);
-                //return promiseRef.Result.Task.ContinueWith(tr =>
-                //{
-                //    if (tr.Result is Status.Success)
-                //    {
-                //        var resultMsg = tr.Result as Status.Success;
-                //        if (resultMsg.Status is Terminated &&
-                //            resultMsg.Status.AsInstanceOf<Terminated>().ActorRef.Path == target.Path)
-                //            return SetThrottleAck.Instance;
-                //        if (resultMsg.Status is SetThrottleAck)
-                //        {
-                //            internalTarget.Tell(new Unwatch(target, promiseRef));
-                //        }
-                //        return SetThrottleAck.Instance;
-                //    }
-                //    else
-                //    {
-                //        internalTarget.Tell(new Unwatch(target, promiseRef));
-                //       return SetThrottleAck.Instance;
-                //    }
-                //}, TaskContinuationOptions.ExecuteSynchronously);
+                var internalTarget = target.AsInstanceOf<IInternalActorRef>();
```

Comment:
Implemented the TODO, which included the fix needed to make the `SurviveNetworkInstabilitySpec` work as expected